evidence/reactor_test.go

```diff
@@ -270,6 +270,16 @@ func makeAndConnectReactorsAndPools(config *cfg.Config, stateStores []sm.Store)
 
 	}, p2p.Connect2Switches)
 
+	// stop every switch at the end of the test
+	t.Cleanup(func() {
+		for _, reactor := range reactors {
+			for _, peer := range reactor.Switch.Peers().List() {
+				reactor.Switch.StopPeerGracefully(peer)
+			}
+		}
```

Comment:
I don't see why you can't just stop the switches (and the reactors) together. Ideally you want all resources allocated in this function to be cleaned up here so the developer writing the tests doesn't have to worry about it

Reply:
Because stopping `reactor` in the `evidence/reactor_test` cause the `data race` issue. Haven't figured out why we cannot stop it at the end of the test. Perhaps there are some issues in the evidence reactor, so we couldn't stop the reactor in tests.

Reply:
Now is able to close the reactor.Switch after close MultiplexTransport.

Reply:
fix `data race` by changing the evidence reactor function:
https://github.com/tendermint/tendermint/pull/9758/files#diff-65427dd576a958246d5324a19b183357a8859fc49bc9d210b1f3d78a980e184bR167